recipes/qcustomplot/all/conanfile.py

```diff
@@ -39,19 +40,19 @@ def config_options(self):
     def configure(self):
         if self.options.shared:
             self.options.rm_safe("fPIC")
-        # FIXME: we shouldn't have to force shared in qt, but config file
-        #        generated by conan in qt static is likely broken, or maybe env vars.
-        self.options["qt"].shared = True
 
     def requirements(self):
-        if Version(self.version) >= "2.0.0":
-            self.requires("qt/6.4.1")
-        else:
-            self.requires("qt/5.15.7")
+        self.requires("qt/6.4.2", transitive_headers=True, transitive_libs=True)
+
         if self.options.with_opengl and self.settings.os == "Windows":
             self.requires("opengl/system")
 
+    def build_requirements(self):
+        self.tool_requires("qt/<host_version>")
+
     def validate(self):
+        if self.settings.os == "Macos":
+            raise ConanInvalidConfiguration(f"{self.ref} Macos not supported at this moment")
```

Comment:
Let me check this on Monday, maybe we can help keep macos support seeing the error the bot reported :)

Reply:
The error is actually in qt recipe, which does not properly define frameworks needed by qtgui. It could be solved by my other qt PR, so let's wait for it to be merged before retrying this one

Reply:
Need some help on how to proceed with this after https://github.com/conan-io/conan-center-index/pull/23571 was abandoned.